openg2p_registration/models/registration_service.py

```diff
@@ -9,10 +10,77 @@ class RegistrationService(models.Model):
 
     def create_registration_for_single_submission(self, odk_data):
         temp = {}
+        # current program id of a registration
+        current_program_id = int(
+            odk_data["program_ids"][len(odk_data["program_ids"]) - 1]
+        )
+        program_obj = self.env["openg2p.program"].search(
+            [("id", "=", current_program_id)]
+        )
         self.renaming_submission_data_fields(temp, odk_data)
-        self.create_fields_for_registration(temp)
+        autodedup_field = program_obj.autodedup_field
+        action = program_obj.action
+        stage_name = program_obj.stage_name
+        if autodedup_field == "kyc":
+            kycdup_list = (
+                self.env["openg2p.registration"]
+                .search([("kyc_id", "=", temp["bban"])])
+                .ids
+            )
 
-    def create_fields_for_registration(self, temp):
+            if len(kycdup_list) != 0:
+                if action == "merge":
+                    # merging the existing ones with new one and passing the existing_id and current_id as arguments
+                    self.merge_registrations(temp, kycdup_list[0])
+                elif action == "del_old":
+                    # deleting the old registration
+                    duplicate_kyc = self.env["openg2p.registration"].search(
+                        [("kyc_id", "=", temp["bban"])]
```

Comment:
Why are we searching kyc ID with BBAN field?

Reply:
Already data got collected in odk central as "bban".